R/cleanifyString.r

```diff
@@ -15,22 +25,36 @@ cleanifyString.character <- function(x, removeNonAlphaNum = TRUE, replaceSpecial
 
   return(x)}
 
-cleanifyString.factor <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE, oldEncoding = NULL, ...) {
+### ----------------------------------------------------------------------------
+
+cleanifyString.factor <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE,
+                                  oldEncoding = NULL, ...) {
+  lapply(c(removeNonAlphaNum, replaceSpecialChars, oldEncoding),
+         checkmate::assert_logical, len = 1)
+
   old <- list(var = as.character(x), lev = levels(x))
   new <- lapply(old, cleanifyString,
-                removeNonAlphaNum = removeNonAlphaNum, replaceSpecialChars = replaceSpecialChars, oldEncoding = oldEncoding, ...)
+                removeNonAlphaNum = removeNonAlphaNum, replaceSpecialChars = replaceSpecialChars,
+                oldEncoding = oldEncoding, ...)
 
   #browser()
   varNew <- factor(x = new[[1]], levels = unique(new[[2]]))
 
   return(varNew)}
 
-cleanifyString.data.frame <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE, oldEncoding = NULL, ...) {
+### ----------------------------------------------------------------------------
+
+cleanifyString.data.frame <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE,
+                                      oldEncoding = NULL, ...) {
+  lapply(c(removeNonAlphaNum, replaceSpecialChars, oldEncoding),
+         checkmate::assert_logical, len = 1)
+  x <- makeDataFrame(x)
```

Comment:
I think, `x <- makeDataFrame(x)` should be omitted. If `x` were something else (a tibble, for example), the S3 method `cleanifyString.data.frame` would not even be called.